brave/src/main/java/brave/propagation/SamplingFlags.java

```diff
@@ -3,90 +3,100 @@
 import brave.internal.Nullable;
 
 //@Immutable
-public abstract class SamplingFlags {
-  public static final SamplingFlags EMPTY = new SamplingFlagsImpl(null, false);
-  public static final SamplingFlags SAMPLED = new SamplingFlagsImpl(true, false);
-  public static final SamplingFlags NOT_SAMPLED = new SamplingFlagsImpl(false, false);
-  public static final SamplingFlags DEBUG = new SamplingFlagsImpl(true, true);
```

Comment:
intellij showed warnings about classloader deadlock by referencing a subclass ctor in a super. the changes here not only make things smaller, but also remove that possibility.